raft/node.go

```diff
@@ -49,43 +49,90 @@ func (a *SoftState) equal(b *SoftState) bool {
 // Ready encapsulates the entries and messages that are ready to read,
 // be saved to stable storage, committed or sent to other peers.
 // All fields in Ready are read-only.
+//
+// Handling the Ready can be done with or without performance optimizations.
```

Comment:
i feel we should move this explanation to https://github.com/etcd-io/etcd/blob/master/raft/README.md and add a link here.